pkg/kubelet/kubelet.go

```diff
@@ -2931,6 +2931,12 @@ func (kl *Kubelet) setNodeReadyCondition(node *api.Node) {
 		}
 	}
 
+	// Record any soft requirements that were not met in the container manager.
+	status := kl.containerManager.Status()
+	if status.SoftRequirements != nil {
+		newNodeReadyCondition.Message = fmt.Sprintf("%s. %s", newNodeReadyCondition.Message, status.SoftRequirements.Error())
```

Comment:
The reason is either `NodeReady` or `NodeNorReady`. In the future, we will make this a hard requirement.